### ISSUE TITLE
Drop support for EOLed Python versions

### ISSUE THREAD

**Issue by antonagestam** [dpkp/kafka-python#2404]
As this library has been struggling keeping up with maintenance (correct me if I'm wrong about that), would it make sense to drop support for all the Python versions that has reached EOL?

Especially seeing the recent issues with six was quite surprising, as Python 2 has now been a dead language for close to 4 years.

Specifically I propose to drop support for versions 2.7, 3.4, 3.5, 3.6, and 3.7.

**Comment by wbarnha** [comment]
I intend to drop support for the older versions of Python after the next release, when I figure out what action to take with PyPi.

**Comment by a-detiste** [comment]
I can provide a PR

**Comment by wbarnha** [comment]
Correct, I don't have any updates until the owner gets back to me. I apologize for the inconvenience.

**Comment by wbarnha** [comment]
`kafka-python-ng==2.0.3` is now available!
```bash
pip3 install kafka-python-ng
```

**Comment by antonagestam** [comment]
@wbarnha Interesting. Would it be possible to release that off of this repository instead of the fork?

**Comment by wbarnha** [comment]
Not yet. If I were to release off this repository, it would create the initial impression to some people that `kafka-python` itself is getting updates into PyPi, which is not true. Additionally, I would have to go back, re-tag, re-release, and rerun the release workflows. I'm very cautious about messing up a project that is getting an estimate of 2.4M downloads per week (as of 3/10/2024) and I don't want to recreate the stressful mistakes I had when I was more actively maintaining https://github.com/faust-streaming/faust (which is actually how I ended up here, because I was trying to save this project as a dependency of `aiokafka`).

**Comment by dpkp** [comment]
Closing issue as it became a bit more about fork. I suspect kafka-python will retain py2.7 support until the next major release (3.x). I Expect to release another minor version with KIP-98 transactional support (likely 2.2).
